File: Tutorial5-2Dagger-Android-SubComponents2/src/main/java/com/example/tutorial5_2dagger_android_subcomponents2/di/ToastMakerModule.java
package com.example.tutorial5_2dagger_android_subcomponents2.di;


import android.content.Context;


import com.example.tutorial5_2dagger_android_subcomponents2.ToastMaker;

import dagger.Module;
import dagger.Provides;


/**
 * <p></p>
 * Sub component and @Provides annotated methods inside module CAN NOT not have different scopes.
 * <p></p>
 * IMPORTANT: Both can be un-scoped but cannot have with different scopes.
 */
@Module
public abstract class ToastMakerModule {

    @ActivityScope
    @Provides
    static ToastMaker provideToastMaker(Context context) {
        System.out.println("ToastMakerModule context: " + context);
        return new ToastMaker(context);
    }

    // This method gets Application Context and returns it
/*    @Provides
    static Context provideContext(Application application) {
        return application.getApplicationContext();
    }*/

}

File: Tutorial5-2Dagger-Android-SubComponents2SeedInstance/src/main/java/com/example/tutorial5_2dagger_android_subcomponents2/di/ToastMakerModule.kt
package com.example.tutorial5_2dagger_android_subcomponents2.di

import android.content.Context
import com.example.tutorial5_2dagger_android_subcomponents2.model.ToastMaker
import dagger.Module
import dagger.Provides

/**
 *
 * Sub component and @Provides annotated methods inside module CAN NOT not have different scopes.
 *
 * IMPORTANT: Both can be un-scoped but cannot have with different scopes.
 */
@Module
abstract class ToastMakerModule {

    @Module
    companion object {

        @JvmStatic
        @ActivityScope
        @Provides
        fun provideToastMaker(context: Context): ToastMaker {
            return ToastMaker(context)
        }

    }

    // This method gets Application Context and returns it

/*    @Provides
    static Context provideContext(Application application) {
        return application.getApplicationContext();
    }*/
}